### PR TITLE
Add start of front-end JSON API.

### PR DESCRIPTION
**Pull request by hodgestar**
This ticket is just to get the initial framework and maybe a few example methods in place. Specific API feature requests can come later (although of course suggestions for API methods are welcome here too -- I'll likely create tickets for them).

### CODE REVIEW

**Comment by hodgestar** [comment]
PR created so that people can follow along. Not ready for review.

**Comment by jerith** [comment]
Looks reasonable, but I don't know enough about the Twisted auth stuff to review that confidently.

File: go/api/go_api/go_api.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    def _format_conversation_list(self, convs):
+        return [ConversationType.format_conversation(c) for c in convs]
+
+    @signature(returns=List("List of campaigns.",
+                            item_type=CampaignType()))
+    def jsonrpc_campaigns(self):
+        """List the campaigns a user has access to."""
+        return [CampaignType.format_campaign({
+            'key': self.user_account_key,
+            'name': u"Your Campaign",
```
Comment (justinvdm) [inline go/api/go_api/go_api.py:73]: Is `Your Campaign` a stub?

Comment (hodgestar) [inline go/api/go_api/go_api.py:73]: Sort of. What we call `accounts` currently will soon become campaigns and each user will (potentially) have access to multiple campaigns. This method returns the correct answer for now (i.e. that a user has one campaign and its key in the `user_account_key`). The name is made up (although I guess `Your Campaign` is not a bad name for the default campaign in an account).

**Comment by justinvdm** [comment]
I know next to nothing about Twisted auth, but I'm :+1: on the way we are doing the API.

**Pull request merged**